server/command.go

```diff
@@ -0,0 +1,233 @@
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to store the new link, please try again.")
+	}
+
+	p.restart()
+
+	return &model.CommandResponse{
+		ResponseType: model.CommandResponseTypeEphemeral,
+		Text:         "The msteams channel is now linked to this mattermost channel",
+		Username:     botDisplayName,
+		IconURL:      msteamsLogoURL,
+	}, nil
+}
+
+func (p *Plugin) executeUnlinkCommand(c *plugin.Context, args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+	channel, appErr := p.API.GetChannel(args.ChannelId)
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to get the current channel information.")
+	}
+
+	canLinkChannel := channel.Type == model.ChannelTypeOpen && p.API.HasPermissionToChannel(args.UserId, args.ChannelId, model.PermissionManagePublicChannelProperties)
+	canLinkChannel = canLinkChannel || (channel.Type == model.ChannelTypePrivate && p.API.HasPermissionToChannel(args.UserId, args.ChannelId, model.PermissionManagePrivateChannelProperties))
+	if !canLinkChannel {
+		return cmdError(args.ChannelId, "Unable to unlink the channel, you has to be a channel admin to unlink it.")
+	}
+
+	channelsLinkedData, appErr := p.API.KVGet("channelsLinked")
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to get the linked channels information, please try again.")
+	}
+	var channelsLinked map[string]ChannelLink
+	json.Unmarshal(channelsLinkedData, &channelsLinked)
+	_, ok := channelsLinked[channel.TeamId+":"+channel.Id]
+	if !ok {
+		return cmdError(args.ChannelId, "Link doesn't exists.")
+	}
+
+	delete(channelsLinked, channel.TeamId+":"+channel.Id)
+	channelsLinkedData, err := json.Marshal(channelsLinked)
+	if err != nil {
+		return cmdError(args.ChannelId, "Unable to store the new link, please try again.")
+	}
+
+	appErr = p.API.KVSet("channelsLinked", channelsLinkedData)
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to store the new link, please try again.")
+	}
+
+	p.restart()
+
+	return &model.CommandResponse{
+		ResponseType: model.CommandResponseTypeEphemeral,
+		Text:         "The msteams channel is no longer linked to this mattermost channel",
+		Username:     botDisplayName,
+		IconURL:      msteamsLogoURL,
+	}, nil
+}
+
+func (p *Plugin) executeShowCommand(c *plugin.Context, args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+	channel, appErr := p.API.GetChannel(args.ChannelId)
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to get the current channel information.")
+	}
+
+	channelsLinkedData, appErr := p.API.KVGet("channelsLinked")
+	if appErr != nil {
+		return cmdError(args.ChannelId, "Unable to get the linked channels information, please try again.")
+	}
+	var channelsLinked map[string]ChannelLink
+	json.Unmarshal(channelsLinkedData, &channelsLinked)
+	link, ok := channelsLinked[channel.TeamId+":"+channel.Id]
+	if !ok {
+		return cmdError(args.ChannelId, "Link doesn't exists.")
+	}
```

Comment:
There's some duplication among the command handlers for loading the channel links. Thoughts on extracting out that logic for reuse?

Reply:
good point

Reply:
I just removed the reading from the KV store; in theory, it is always in memory, so it shouldn't be a problem.